CONTRIBUTING.md

```diff
@@ -0,0 +1,88 @@
+
+This can be done by looking in the top-right corner of the repository page and click "Fork".
+![fork](assets/images/fork.jpg)
+
+The next step is to clone your newly forked repository onto your local workspace. This can be done by going to your newly forked repository, which should be at `https://github.com/USERNAME/wildfly-elytron`. 
+
+Then, there will be a green button that says "Code". Click on that and copy the URL.
+
+![clone](assets/images/clone.png)
+
+Then, in your terminal, paste the following command:
+`git clone [URL]`
+Be sure to replace [URL] with the URL that you copied.
+
+Now you have the repository on your computer!
+
+## Issues
+The WildFly Elytron project uses JIRA to manage issues. All issues can be found [here](https://issues.redhat.com/projects/ELY/issues). 
+
+To create a new issue, comment on an existing issue, or assign an issue to yourself, you'll need to first [create a JIRA account](https://issues.redhat.com/).
+
+
+### Good First Issues
+Want to contribute to the WildFly Elytron project but aren't quite sure where to start? Check out our issues with the `good-first-issue` label. These are a triaged set of issues that are great for getting started on our project. These can be found [here](https://issues.redhat.com/issues/?filter=12383825). 
+
+Once you have selected an issue you'd like to work on, make sure it's not already assigned to someone else. Then, remember to assign it to yourself, by clicking on "Assign to me", to prevent someone else from also working on the same issue.
+
+![jira](assets/images/jira.png)
+
+It is recommend that you use a separate branch for every Pull Request to submit. To keep things straightforward and memorable, you should try to name the branch the JIRA issue number. Doing so also allows your to prevent conflicts in the future when working on other issues. For example, for ELY-2127, you should also name the branch ELY-2127.
+
+## Setting up your Developer Environment
+You will need:
+
+* JDK 11
+* Git
+* Maven 3.3.9 or later
+* An [IDE](https://en.wikipedia.org/wiki/Comparison_of_integrated_development_environments#Java)
+(e.g., [IntelliJ IDEA](https://www.jetbrains.com/idea/download/), [Eclipse](https://www.eclipse.org/downloads/), etc.)
+
+First `cd` to the directory where you cloned the project (eg: `cd wildfly-elytron`)
+
+Add a remote ref to upstream, for pulling future updates.
+For example:
+
+```
+git remote add upstream https://github.com/wildfly-security/wildfly-elytron
+```
+To build `wildfly-elytron` run:
+```bash
+mvn clean install
+```
+
+To skip the tests, use:
+
+```bash
+mvn clean install -DskipTests=true
+```
+
+To run only a specific test, use:
+
+```bash
+mvn clean install -Dtest=TestClassName
+```
+For more information, including details on how WildFly Elytron is integrated in WildFly Core and WildFly, check out our [developer guide](https://wildfly-security.github.io/wildfly-elytron/getting-started-for-developers/).
+
+## Coding Guidlines
```

Comment:
Typo in Guidelines word. Btw I would maybe call this Contribution Guidelines. As it is not about the actual java code but about style of PRs and commit messages.

Reply:
Makes sense, changed